### PR TITLE
Umar2

### PR DESCRIPTION
**Pull request by khan168**

### CODE REVIEW

File: server/controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        }
+    }catch(err){
+            res.status(400)
+            res.json({ error: err });
+    }
+}
+
+// @ desc       sign up user
+// @ router     router /api/user/signup
+const  register = async (req, res,next) => {
```
Comment (nhan-ngo-usf) [inline server/controllers/authController.js:38]: what is next use for???

Comment (khan168) [reply server/controllers/authController.js:38]: for no reason, I forgot removing it. To make sure it works, I tested everything by removing it. Review everything apart from that and let me know

File: server/controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        if(existinguser){
+            return res.status(400).json({error:"Invalid credentials"})
+        }
+        const hashedPassword = await bcrypt.hash(password, 8);
+        const user = await User.create({
+            name,
+            email,
+            password:hashedPassword
+        })
+        if(user){
```
Comment (nhan-ngo-usf) [inline server/controllers/authController.js:56]: you already handle invalid input in validator middleware so no need for if here

File: package.json [hunk 3]
```diff
@@ -4,13 +4,22 @@
...
+    "bcrypt": "^5.1.0",
+    "bcryptjs": "^2.4.3",
+    "body-parser": "^1.20.1",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "express-async-handler": "^1.2.0",
+    "express-validator": "^6.14.2",
+    "jsonwebtoken": "^8.5.1",
+    "mongoose": "^6.7.3",
+    "nodemon": "^2.0.20",
```
Comment (nhan-ngo-usf) [inline package.json:22]: devDependencies

Comment (khan168) [reply package.json]: I only moved nodemon

**Review by nhan-ngo-usf — APPROVED** [review]
Approved request

**Pull request merged**